traits/tests/test_float.py

```diff
@@ -18,6 +18,24 @@
 from traits.testing.optional_dependencies import numpy, requires_numpy
 
 
+class IntegerLike:
+    def __init__(self, value):
+        self._value = value
+
+    def __index__(self):
+        return self._value
+
+
+# Python versions < 3.8 don't support conversion of something with __index__
+# to float.
+try:
+    float(IntegerLike(3))
+except TypeError:
+    float_accepts_index = False
+else:
+    float_accepts_index = True
```

Comment:
Why not just use `sys.version_info`?

Reply:
I tend to follow the general rule of checking for features directly where possible, rather than using the version number as a proxy for feature presence / absence.